### PR TITLE
feat(module:transfer): add transfer component

### PR DESCRIPTION
**Pull request by cipchk**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/components/ng-zorro-antd.module.ts [hunk 1]
```diff
@@ -118,6 +119,7 @@ export { NzBackTopModule } from './back-top/nz-back-top.module';
 export { NzAffixModule } from './affix/nz-affix.module';
 export { NzAnchorModule } from './anchor/nz-anchor.module';
 export { NzAvatarModule } from './avatar/nz-avatar.module';
+export { NzTransferModule } from './transfer/nz-transfer.module';
```
Comment (vthinkxie) [inline src/components/ng-zorro-antd.module.ts]: forget to export TransferComponent

Comment (cipchk) [reply src/components/ng-zorro-antd.module.ts]: fixed

File: src/components/transfer/nz-transfer-search.component.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+
+@Component({
+    selector: 'nz-transfer-search',
+    template: `
+    <input nz-input type="text" [(ngModel)]="value" (ngModelChange)="_handle()"
```
Comment (vthinkxie) [inline src/components/transfer/nz-transfer-search.component.ts]: there is no need to add type="text" as default attribute

File: src/showcase/nz-demo-transfer/nz-demo-transfer.html [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+        </tr>
+        <tr>
+          <td>nzListStyle</td>
+          <td>两个穿梭框的自定义样式，以<code>ngStyle</code>写法标题</td>
+          <td>object</td>
+          <td></td>
+        </tr>
+        <tr>
+          <td>nzItemUnit</td>
+          <td>标题项目单位</td>
```
Comment (vthinkxie) [inline src/showcase/nz-demo-transfer/nz-demo-transfer.html]: not clear enough
`单数单位` maybe better?

File: src/showcase/nz-demo-transfer/nz-demo-transfer.html [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+          <td>string</td>
+          <td>项目</td>
+        </tr>
+        <tr>
+          <td>nzItemsUnit</td>
+          <td>标题项目单位</td>
```
Comment (vthinkxie) [inline src/showcase/nz-demo-transfer/nz-demo-transfer.html]: not clear enough
`复数单位` maybe better?

File: src/components/transfer/nz-transfer-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+    ngDoCheck(): void {
+      const change = this._listDiffer.diff(this._list);
+      if (change) {
+        this.updateCheckStatus();
+      }
+    }
+
+    _handleSelect(item: TransferItem) {
+      item.checked = !item.checked;
```
Comment (vthinkxie) [inline src/components/transfer/nz-transfer-list.component.ts]: handle with disabled state?

**Comment by cipchk** [comment]
Button icon don't alignment, i do not know how it is caused?
jpg:

tim 20171117114331

do you have any advice?

File: src/components/transfer/nz-transfer-list.component.ts (hunk 5 above)
Comment (cipchk) [reply src/components/transfer/nz-transfer-list.component.ts]: no, originally just rely `nz-checkbox`, thx.

**Comment by vthinkxie** [comment]
@cipchk it seems ok on mac computer.

**Pull request merged**